File: stsservice/poc.go
package main

import (
"github.com/aws/aws-sdk-go/aws/session"
"github.com/aws/aws-sdk-go/service/sts"
"github.com/aws/aws-sdk-go/service/sts/stsiface"
"fmt"
)

const (
	AwsDefaultRegion   = "us-west-1"
	AwsAccessKeyId     = "access-key-id"
	AwsSecretAccessKey = "secret-access-key"
)

func setAwsConfig() {
	// do the credential setting here
	//os.Setenv("AWS_ACCESS_KEY_ID", AwsAccessKeyId)
	//os.Setenv("AWS_SECRET_ACCESS_KEY", AwsSecretAccessKey)
}

func getStsClient() stsiface.STSAPI {
	sess := session.Must(session.NewSession())
	client := sts.New(sess)
	return client
}

func getSessionToken(client stsiface.STSAPI) (*sts.GetSessionTokenOutput, error) {
	sessionToken, err := client.GetSessionToken(&sts.GetSessionTokenInput{
	})
	if err != nil {
		fmt.Println(err)
		return nil, err
	}
	return sessionToken, nil
}

func main() {
	client := getStsClient()
	sessionToken, err := getSessionToken(client)
	if err != nil {
		fmt.Println(err)
		return
	}
	fmt.Println(sessionToken)
}

File: stsservice/poc.py
import boto3

AWS_DEFAULT_REGION = 'us-west-1'
AWS_ACCESS_KEY_ID = 'access-key-id'
AWS_SECRET_ACCESS_KEY = 'secret-access-key'

def set_aws_config():
    # do the credential setting here
    #os.environ['AWS_ACCESS_KEY_ID'] = AWS_ACCESS_KEY_ID
    #os.environ['AWS_SECRET_ACCESS_KEY'] = AWS_SECRET_ACCESS_KEY
    pass

def get_sts_client():
    set_aws_config()
    session = boto3.session.Session()
    client = session.client('sts')
    return client

def main():
    sts_client = get_sts_client()
    session_token = sts_client.get_session_token()
    print(session_token)

if __name__ == '__main__':
    main()
